plugin.video.cbc/lib/scraper.py

```diff
@@ -0,0 +1,192 @@
+   shows = re.compile('<a href="(.+?)">(.+?)</a>', re.DOTALL).findall(html)
+   shows.append(('/player/news','News'))
+   shows.append(('/player/Sports','Sports'))
+   shows.append(('/player/news/TV%20Shows/The%20National/Latest%20Broadcast','The National - Latest Broadcast'))
+
+   for url, name in shows:
+      ilist = self.addMenuItem(name,'GC', ilist, url+'|'+name, self.addonIcon, self.addonFanart, infoList, isFolder=True)
+   return(ilist)
+
+
+ def getAddonCats(self,lurl,ilist):
+   infoList = {}
+   url  = uqp(lurl)
+   xurl = url.split('|',1)
+   gcurl = urllib.quote(xurl[0])
+   gcurl = gcurl.replace('%3A',':',1)
+   catname = xurl[1]
+   html  = self.getRequest('http://www.cbc.ca%s' % gcurl)
+   try:    
+           html = re.compile('<section class="category-subs full">(.+?)</section', re.DOTALL).search(html).group(1)
+           shows = re.compile('a href="(.+?)">(.+?)<', re.DOTALL).findall(html)
+   except:
+      try:
+           html = re.compile('<section class="category-content full">(.+?)</section', re.DOTALL).search(html).group(1)
+           ilist = self.getAddonEpisodes(lurl, ilist)
+           return(ilist)
+      except:
+           html = re.compile('<section class="section-cats full">(.+?)</section', re.DOTALL).search(html).group(1)
+           shows = re.compile('a href="(.+?)">(.+?)<', re.DOTALL).findall(html)
+        
+
+   for url, name in shows:
+      ilist = self.addMenuItem(name,'GE', ilist, url+'|'+catname, self.addonIcon, self.addonFanart, infoList, isFolder=True)
+   return(ilist)
+
+  
+
+ def getAddonShows(self,url,ilist):
+   return(ilist)
+
+
+ def getAddonEpisodes(self,url,ilist):
+   xbmcplugin.setContent(int(sys.argv[1]), 'episodes')
+   self.defaultVidStream['width']  = 960
+   self.defaultVidStream['height'] = 540
+
+   url   = uqp(url)
+   geurl, sname = url.split('|',1)
+   geurl = urllib.quote(geurl)
+   geurl = geurl.replace('%3A',':',1)
+   meta = self.getAddonMeta()
+   try:    i = len(meta[sname])
+   except: meta[sname]={}
+   html  = self.getRequest('http://www.cbc.ca%s' % geurl)
+   html  = re.compile('<section class="category-content full">(.+?)</section', re.DOTALL).search(html).group(1)
+   if '<li class="active">' in html:
+        return(self.getAddonShows(url, ilist))
+
+   epis  = re.compile('<a href="(.+?)" aria-label="(.+?)".+?src="(.+?)".+?</a', re.DOTALL).findall(html)
+   numShows = len(epis)
+   pDialog = xbmcgui.DialogProgress()
+   pDialog.create(self.addonName, self.localLang(30101))
+   pDialog.update(0)
+   dirty = False
+   for i,(url,name,img) in list(enumerate(epis, start=1)):
+     name = name.decode('utf-8','replace')
+     try:
+         (name, img, infoList) = meta[sname][url] 
+     except:
+         infoList={}
+         try:
+            html = self.getRequest(url)
+            try:
+                plot = re.compile('<meta name="description" content="(.+?)"', re.DOTALL).search(html).group(1)
+            except:
```

Comment:
If I recall, I just added on to fix a website change. Probably should go back and make this a bit neater in the next update. Thanks.